sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -76,7 +76,7 @@ trait NonLeafStatementExec extends CompoundStatementExec {
       val df = Dataset.ofRows(session, statement.parsedPlan)
       df.schema.fields match {
         case Array(field) if field.dataType == BooleanType =>
-          df.limit(2).collect() match {
+          df.limit(1).collect() match {
```

Comment:
Is this really a bug? if the dataframe contains more than one row, then df.limit(2).collect() would fail, but limit(1) can pass. cc @davidm-db

Reply:
Oh, you're right. Indeed, I misunderstood this logic. Let me close it.

Reply:
@panbingkun yes, what Allison said is correct, if there are multiple rows in the result, then we don't want to return `true`.